src/main/java/Duke.java

```diff
@@ -1,10 +1,174 @@
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
+        /*String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        System.out.println("Hello from\n" + logo);*/
+        String greetings = " Hello! I'm Duke\n" + " What can I do for you?\n";
+        System.out.println(greetings);
+        Task[] Tasks = new Task[100];
+        Scanner sc= new Scanner(System.in); //System.in is a standard input stream
+        boolean flag = true;
+        int maxlength = 0;
+        for(int j = 0; j < 100 && flag;){
+            String str= sc.nextLine();
+            if(str.equals("bye")){
+                str = "Bye. Hope to see you again soon!";
+                printString(str);
+                flag = false;
+            }else if(str.equals("list")){
+                printList(Tasks, maxlength);
+            }else if(str.contains("todo")) {
+                if(str.length() - 5 + 9> maxlength) maxlength = str.length() - 5 + 9;//- length of "todo " + "1. [X][X]"
+                str = str.substring(5, str.length());
+                Todo t = new Todo(str);
+                addToList(t, Tasks, j);
+                j++;
+                printTask(t, j);
+            }else if(str.contains("deadline")) {
+                if(!str.contains("/by")){
+                    printString("bad format");
+                }else {
+                    String time = str.substring(str.indexOf("/") + 3, str.length());
+                    if (str.length() - 9 + 11 > maxlength) maxlength = str.length() - 9 + 11;//- length of "deadline " + "1. [X][X]" + "(xx:" + ")"
+                    Deadline t = new Deadline(str.substring(9, str.indexOf("/") - 1), time);
+                    addToList(t, Tasks, j);
+                    j++;
+                    printTask(t, j);
+                }
+            }else if(str.contains("event")) {
+                if(!str.contains("/at")){
+                    printString("bad format");
+                }else {
+                    String time = str.substring(str.indexOf("/") + 3, str.length());
+                    if (str.length() - 6 + 11 > maxlength) maxlength = str.length() - 6 + 11;
+                    Event t = new Event(str.substring(6, str.indexOf("/") - 1), time);
+                    addToList(t, Tasks, j);
+                    j++;
+                    printTask(t, j);
+                }
+            }else if(str.contains("done")) {
+                String numberOnly = str.replaceAll("[^0-9]", "");
+                int num = Integer.parseInt(numberOnly);
+                if (num <= j) {
+                    Tasks[num - 1].setDone(true);
+                    printDone(Tasks[num - 1]);
+                } else {
+                    str = "not assigned yet";
```

Comment:
The reuse of the str String variable may be non-intuitive - it initially refers to the command input by the user, and after re-assignment, it became the message back to the user. Consider using separate String names such as userCommand and returnMessage. Otherwise, just parse "not assigned yet" into printString()